packages/eslint-plugin/src/rules/confusing-non-null-assertion-like-not-equal.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  AST_NODE_TYPES,
+  AST_TOKEN_TYPES,
+  TSESLint,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import * as util from '../util';
+
+type MessageIds = 'notNeedInEqualTest' | 'confusing';
+
+export default util.createRule<[], MessageIds>({
```

Comment:
nit - don't need to declare `MessageIds` separately if there are no options

```suggestion
export default util.createRule({
```